src/Search/ReverseGeocoder_P.cpp

```diff
@@ -258,66 +267,159 @@ QVector<std::shared_ptr<const OsmAnd::ReverseGeocoder::ResultEntry>> OsmAnd::Rev
                                                (const std::shared_ptr<const OsmAnd::Road>& road) -> bool
                                                {
                                                    return !road->captions.isEmpty();
                                                });
-    if (roads.isEmpty())
-        roads = roadLocator->findNearestRoads(searchPoint31, STOP_SEARCHING_STREET_WITHOUT_MULTIPLIER_RADIUS * 10, OsmAnd::RoutingDataLevel::Detailed,
+    if (dataObjects.isEmpty())
+        dataObjects = roadLocator->findNearestRoads(searchPoint31, STOP_SEARCHING_STREET_WITHOUT_MULTIPLIER_RADIUS * 10, OsmAnd::RoutingDataLevel::Detailed,
                                               [this]
                                               (const std::shared_ptr<const OsmAnd::Road>& road) -> bool
                                               {
                                                   return !road->captions.isEmpty();
                                               });
-    
+            
+    if (list.empty())
+        list = {};
+
+    for (auto r : dataObjects)
+    {
+        if (r.first->points31.size() > 0)
+        {
+            std::shared_ptr<RouteSegmentPoint> road = nullptr;
+            for (int j = 1; j < r.first->points31.size(); j++)
+            {
+                std::pair<int, int> pr = Utilities::getProjectionPoint31(searchPoint31.x, searchPoint31.y, r.first->points31[j-1].x, r.first->points31[j-1].y, r.first->points31[j].x, r.first->points31[j].y);
+                double currentsDistSquare = Utilities::squareDistance31(pr.first, pr.second, searchPoint31.x, searchPoint31.y);
+                if (road == nullptr || currentsDistSquare < road->dist)
+                {
+                    SHARED_PTR<RouteDataObject> ro = std::make_shared<RouteDataObject>();
+                    std::vector<uint32_t> pointsX = {};
+                    std::vector<uint32_t> pointsY = {};
+                    for (int k = 0; k < r.first->points31.size(); k++)
+                    {
+                        pointsX.push_back(r.first->points31[k].x);
+                        pointsY.push_back(r.first->points31[k].y);
+                    }
+                    ro->pointsX = pointsX;
+                    ro->pointsY = pointsX;
+                    for (uint32_t key : r.first->captions.keys())
+                    {
+                        ro->names[(int)key] = r.first->captions[key].toStdString();
+                    }
+                    
+                    road = std::make_shared<RouteSegmentPoint>(ro, j, sqrt(currentsDistSquare));
+                    road->preciseX = pr.first;
+                    road->preciseY = pr.second;
+                    this->regionsCache[road] = r.first->section->name;
+                }
+            }
+            
+            if (road != nullptr)
+            {
+                float prio = ctx->config->router->defineDestinationPriority(road->road);
+                if (prio > 0)
```

Comment:
prio in most cases equal 1
https://github.com/osmandapp/OsmAnd-core-legacy/blob/1bb2bd4a5a8a8d6aa713ca1ceb00acb02d78f26f/native/src/generalRouter.cpp#L517

0.5 only if road is tunnel
https://github.com/osmandapp/OsmAnd-resources/blob/43ad7012adc25b23362453ee0c76d2621b84fe37/routing/routing.xml#L885